### ISSUE TITLE
Status of this project?

### ISSUE THREAD

**Issue by tarngerine** [mindreframer/vite_phx#6]
Hi! I've been playing with Vite + Phoenix and read your article (many times lol) and was curious what your current status is with this project!

1. Are you using it in your own project daily?
2. Does it support HMR?
3. In the React example, at a high level, could you describe the reason for both root.html.eex and vite.html.eex? 

New to build stuff and trying to learn, thank you for your effort and sharing this!

**Comment by mindreframer** [comment]
Hey @tarngerine! 

1. I was using it on my last consulting project, with a heavy focus on interactive React.js frontend. It was quite valuable, because it was retaining the browser state that required multiple steps to assemble. This allowed realtime tweaking for layout / design / colours, so the DX to use it was great. The DX to set it up and also explain to team-members was somewhat exotic and it felt like "magic". Magic is a double-edged sword, so I am not 100% sure, that the additional configuration effort is justified for every project. Also, I was trying to keep the option to use serverside rendered HTML for some simpler areas and Reactjs for more interactive screens. It felt more complex than it should be. 

Honestly, I feel like I'm missing something, that will make building good frontend in a simple way possible yet reap the benefits 
 of realtime HRM updates during dev. But maybe I have to keep experimenting more in that direction. 

2. It does
3. root and vite layouts would allow to have the option to use phoenix templates or vite.js logic. 


I also have another phoenix / vite setup, that is simpler in setup, yet goes more direction SPA. I could publish an example, that might give you and other developers some ideas.

**Comment by rshigg** [comment]
@mindreframer yes please do post the SPA setup! I have been trying to figure out how to integrate a vite react SPA with a phoenix backend (for channels and presence) for a while now and that would be a real life-saver.
